hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -0,0 +1,869 @@
+
+var _ genruntime.Reconciler = &AzureDeploymentReconciler{}
+
+type AzureDeploymentReconciler struct {
+	obj                  genruntime.MetaObject
+	log                  logr.Logger
+	recorder             record.EventRecorder
+	ARMClient            armclient.Applier
+	KubeClient           *kubeclient.Client
+	ResourceResolver     *armresourceresolver.Resolver
+	CreateDeploymentName func(obj metav1.Object) (string, error)
+}
+
+func NewAzureDeploymentReconciler(
+	metaObj genruntime.MetaObject,
+	log logr.Logger,
+	armClient armclient.Applier,
+	eventRecorder record.EventRecorder,
+	kubeClient *kubeclient.Client,
+	resourceResolver *armresourceresolver.Resolver,
+	createDeploymentName func(obj metav1.Object) (string, error)) genruntime.Reconciler {
+
+	return &AzureDeploymentReconciler{
+		obj:                  metaObj,
+		log:                  log,
+		recorder:             eventRecorder,
+		ARMClient:            armClient,
+		CreateDeploymentName: createDeploymentName,
+		KubeClient:           kubeClient,
+		ResourceResolver:     resourceResolver,
+	}
+}
+
+func (r *AzureDeploymentReconciler) CreateOrUpdate(ctx context.Context) (ctrl.Result, error) {
+	action, actionFunc, err := r.DetermineCreateOrUpdateAction()
+
+	if err != nil {
+		r.log.Error(err, "error determining create or update action")
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "DetermineCreateOrUpdateActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	result, err := actionFunc(ctx, action)
+	if err != nil {
+		r.log.Error(err, "Error during CreateOrUpdate", "action", action)
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "CreateOrUpdateActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	return result, nil
+}
+
+func (r *AzureDeploymentReconciler) Delete(ctx context.Context) (ctrl.Result, error) {
+	action, actionFunc, err := r.DetermineDeleteAction()
+
+	if err != nil {
+		r.log.Error(err, "error determining delete action")
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "DetermineDeleteActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	result, err := actionFunc(ctx, action)
+	if err != nil {
+		r.log.Error(err, "Error during Delete", "action", action)
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "DeleteActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	return result, nil
+}
+
+func (r *AzureDeploymentReconciler) IsTerminalProvisioningState() bool {
+	state := r.GetResourceProvisioningState()
+	return armclient.IsTerminalProvisioningState(state)
+}
+
+func (r *AzureDeploymentReconciler) GetResourceProvisioningState() armclient.ProvisioningState {
+	return armclient.ProvisioningState(r.obj.GetAnnotations()[ResourceStateAnnotation])
+}
+
+func (r *AzureDeploymentReconciler) GetDeploymentId() (string, bool) {
+	id, ok := r.obj.GetAnnotations()[DeploymentIdAnnotation]
+	return id, ok
+}
+
+func (r *AzureDeploymentReconciler) GetDeploymentIdOrDefault() string {
```

Comment:
Why do we need this method when we have `GetDeploymentId()`? I thought idiomatic Go was to spurn helper methods like this.

Reply:
Yeah in this instance at least this method isn't buying us much - it'd be clearer to just use `GetDeploymentId`, I've made the change.